src/vmm/src/device_manager/persist.rs

```diff
@@ -645,16 +646,74 @@ mod tests {
         let vmm = default_vmm();
         let device_states: DeviceStates =
             DeviceStates::deserialize(&mut buf.as_slice(), &version_map, 2).unwrap();
+        let vm_resources = &mut Default::default();
         let restore_args = MMIODevManagerConstructorArgs {
             mem: vmm.guest_memory().clone(),
             vm: vmm.vm.fd(),
             event_manager: &mut event_manager,
             for_each_restored_device: VmResources::update_from_restored_device,
-            vm_resources: &mut Default::default(),
+            vm_resources,
         };
         let restored_dev_manager =
             MMIODeviceManager::restore(restore_args, &device_states).unwrap();
 
+        let expected_vm_resources = format!(
+            r#"{{
+  "balloon": {{
+    "amount_mib": 123,
+    "deflate_on_oom": false,
+    "stats_polling_interval_s": 1
+  }},
+  "drives": [
+    {{
+      "drive_id": "root",
+      "path_on_host": "{}",
+      "is_root_device": true,
+      "partuuid": null,
+      "is_read_only": true,
+      "cache_type": "Unsafe",
+      "rate_limiter": null,
+      "io_engine": "Sync"
+    }}
+  ],
+  "boot-source": null,
+  "logger": null,
+  "machine-config": {{
+    "vcpu_count": 1,
+    "mem_size_mib": 128,
+    "ht_enabled": false,
+    "track_dirty_pages": false
+  }},
+  "metrics": null,
+  "mmds-config": null,
+  "network-interfaces": [
+    {{
+      "iface_id": "netif",
+      "host_dev_name": "hostname",
+      "guest_mac": "00:00:00:00:00:00",
```

Comment:
@alindima  I'm unsure what to think about this guest_mac being all 0s. It feels like it should actually be null because the guest_mac is None on line 604 but then I saw some code in Net::new_with_tap that seemed to use a ConfigSpace::default() which has all 0s for the mac address so then I figured maybe the all 0s is right.

Reply:
Nice catch! I've done some digging to fond out why that is.
This is all zeros because indeed the guest mac in the config space is initialised with zeros. When the Net device is then saved to snapshot, we don't specifically save the guest mac as configured by the user and we then initialise it from the config space value, which in this case is all zeros.

I think this is a bug so we should open a new issue for it after we merge the PR. It is confusing to have a zeroed MAC when it was not really configured.

Reply:
@alindima You want to wait until after this PR is merged before we create the new issue related to the zeroed out mac address, right? Just going through and making sure I've addressed all outstanding comments.

Reply:
Let's wait until we merge this PR and leave a comment on the same issue with the unaddressed items, like this MAC issue and the fact that the cpu_template, boot source and smt are currently not persisted.